docs/message-guidelines.md

```diff
@@ -0,0 +1,101 @@
+# Message guidelines
+
+## Format
+
+All messages should follow [ROS message description specification](https://docs.ros.org/en/galactic/Concepts/About-ROS-Interfaces.html#background).
+
+The accepted formats are:
+
+- `.msg`
+- `.srv`
+- `.action`
+
+## Default units
+
+All the fields by default have the following units depending on their types:
+
+| type           | default unit  |
+| -------------- | ------------- |
+| distance       | meter (m)     |
+| angle          | radians (rad) |
+| time           | second (s)    |
+| speed          | m/s           |
+| velocity       | m/s           |
+| acceleration   | m/s²          |
+| angular vel.   | rad/s         |
+| angular accel. | rad/s²        |
+
+If the field in a message will have any of these default units, don't add any suffix or prefix denoting the type.
+
+## Non-default units
+
+For non-default units, use following suffixes:
+
+| type     | non-default unit | suffix  |
+| -------- | ---------------- | ------- |
+| distance | nanometer        | `_nm`   |
+| distance | micrometer       | `_um`   |
+| distance | millimeter       | `_mm`   |
+| distance | kilometer        | `_km`   |
+| angle    | degree (deg)     | `_deg`  |
+| time     | nanosecond       | `_ns`   |
+| time     | microsecond      | `_us`   |
+| time     | millisecond      | `_ms`   |
+| time     | minute           | `_min`  |
+| time     | hour (h)         | `_hour` |
+| velocity | km/h             | `_kmph` |
+
+## Message field types
+
+For list of types supported by the ROS interfaces [see here](https://docs.ros.org/en/galactic/Concepts/About-ROS-Interfaces.html#field-types).
```

Comment:
Do we need guidelines for array types ?

My main concern is with _bounded_ arrays which are currently used in some of the `.idl` messages (e.g., https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_planning_msgs/msg/Trajectory.idl#L14).
I do not know the advantage of using these bounds but I know they can cause some issues.
If there is a good reason to have bounded arrays, then it should probably be written in the guideline.
Otherwise I guess the bounds will be removed when converting the messages to `.msg` format.

Reply:
Moved here https://github.com/autowarefoundation/autoware-documentation/pull/94/files#r867909596